src/python_testing/TC_EEVSE_2_6.py

```diff
@@ -0,0 +1,291 @@
+            TestStep("17c", "TH counts all report transactions with an attribute report for the SessionEnergyCharged attribute",
+                     "TH verifies that numberOfReportsReceived >= 1"),
+            TestStep("17d", "If V2X feature is supported on the cluster, TH counts all report transactions with an attribute report for the SessionEnergyDischarged attribute",
+                     "TH verifies that numberOfReportsReceived >= 1"),
+            TestStep("18", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EEVSE.TESTEVENT_TRIGGERKEY and EventTrigger field set to PIXIT.EEVSE.TESTEVENTTRIGGER for EV Plugged-in Test Event Clear",
+                     "Verify DUT responds w/ status SUCCESS(0x00) and event EEVSE.S.E01(EVNotDetected) sent"),
+            TestStep("19", "TH sends TestEventTrigger command to General Diagnostics Cluster on Endpoint 0 with EnableKey field set to PIXIT.EEVSE.TESTEVENT_TRIGGERKEY and EventTrigger field set to PIXIT.EEVSE.TESTEVENTTRIGGER for Basic Functionality Test Event Clear",
+                     "Verify DUT responds w/ status SUCCESS(0x00)"),
+            TestStep("20", "Cancel the subscription to the Device Energy Management cluster",
+                     "The subscription is cancelled successfully"),
+        ]
+
+        return steps
+
+    @async_test_body
+    async def test_TC_EEVSE_2_6(self):
+        self.step("1")
+        # Commission DUT - already done
+
+        # Subscribe to Events and when they are sent push them to a queue for checking later
+        events_callback = EventChangeCallback(Clusters.EnergyEvse)
+        await events_callback.start(self.default_controller,
+                                    self.dut_node_id,
+                                    self.matter_test_config.endpoint)
+
+        self.step("2")
+        feature_map = await self.read_evse_attribute_expect_success(attribute="FeatureMap")
+        logger.info(f"FeatureMap: {feature_map}")
+        has_v2x = feature_map & Clusters.EnergyEvse.Bitmaps.Feature.kV2x
+
+        self.step("3")
+        await self.check_test_event_triggers_enabled()
+
+        self.step("4")
+        sub_handler = ClusterAttributeChangeAccumulator(Clusters.EnergyEvse)
+        await sub_handler.start(self.default_controller, self.dut_node_id,
+                                self.matter_test_config.endpoint,
+                                min_interval_sec=0,
+                                max_interval_sec=10, keepSubscriptions=True)
+
+        def accumulate_reports(wait_time):
+            logging.info(f"Test will now wait {wait_time} seconds to accumulate reports")
+            time.sleep(wait_time)
+
+        self.step("5")
+        await self.send_test_event_trigger_basic()
+
+        self.step("5a")
+        await self.check_evse_attribute("State", Clusters.EnergyEvse.Enums.StateEnum.kNotPluggedIn)
```

Comment:
Test step desc says "after a few seconds" which doesn't seem to be happening here.

Reply:
I've updated the test script - the wait shouldn't be needed.